js/src/react/mapReactData.js

```diff
@@ -82,12 +82,25 @@ dataMappers.element = ({ module, name, props: propsData }) => {
   return element;
 };
 
-// Used to implement `setInput()` and `triggerEvent()` R functions. In case of `triggerEvent()`,
-// we have `argIdx === null` and the returned function just sets the Shiny input to `TRUE`
+// Used to implement `triggerEvent()` R function.
+// The returned function just sets the Shiny input to `TRUE`
 // on every call (this works thanks to `priority: 'event'`).
-dataMappers.input = ({ id, argIdx }) => (
-  (...args) => {
-    const value = argIdx === null ? true : args[argIdx];
+dataMappers.event = ({ id }) => (
+  () => {
+    Shiny.setInputValue(id, true, { priority: 'event' });
+  }
+);
+
+// Used to implement `setInput()` R function.
+dataMappers.input = ({ id, jsAccessor }) => (
+  () => { // used to be (...args) , see below
+    //
+    let value = true;
+    if (jsAccessor !== undefined) {
+      // Needs to use arguments inside eval string, otherwise webpack
+      // won't translate args => arguments
+      value = eval(`arguments${jsAccessor}`); // eslint-disable-line no-eval
+    }
     Shiny.setInputValue(id, value, { priority: 'event' });
   }
 );
```

Comment:
WDYT about this implementation? I dropped the `jsAccessor !== undefined` check - when could it fail?
```js
dataMappers.input = ({ id, jsAccessor }) => {
  const getValue = eval(`(...args) => (args${jsAccessor})`); // eslint-disable-line no-eval
  return (...args) => {
    Shiny.setInputValue(id, getValue(args), { priority: 'event' });
  }
}
```

Reply:
Note: There was a bug in my suggested code above (`...args` instead of `args` in the `eval()`). Now fixed.

Reply:
Very clean!!

I like it a lot and it avoids having to use `arguments` directly as it's no longer inside of a string

I like to safeguard against some stupid bugs, but here it's really not necessary